### DISCUSSION TITLE
Error: Could not resolve allOf schema

### DISCUSSION THREAD

**Discussion by pongohaja** [apideck-libraries/portman#436] [Q&A] [answered]
Hello,
Our OpenAPI specification (OAS) contains an allOf schema. It seems that Portman cannot handle it. However, when importing the OAS as a _Postman Collection_ to Postman as a collection, it is handled correctly.

I suspect that the problem is due to that portman uses an old version of openapi-to-postmanv2
``` npm list openapi-to-postmanv2
<user>
└─┬ @apideck/portman@1.20.1
  └── openapi-to-postmanv2@2.14.1

```
The latest available version is openapi-to-postmanv2@4.4.1. 
I tried to run openapi-to-postmanv2 separately towards my OAS'. 2.14.1 failed while 4.4.1 passed.

**Does anyone know if Portman is to be released with a newer version of openapi-to-postmanv2 that solves the issue?**

See also the attached files.
oas_subtest.yaml.txt - OAS before allOf was introduced
oas_subtest_allof.yaml.txt - OAS with allOf causing the error message.
portman_error_msg.txt

**Symptoms**:
* In generated postman collection: Error: Could not resolve allOf schema
* At command line: "- _Converting OpenApi to Postman Collection
Error while resolving allOf schema:  Error: Could not resolve values for path:"type". They are probably incompatible. Values:
"string"
"object"_
"

**OAS working without allOf**;
_    properties:
        id:
          $ref: '#/components/schemas/UUID'
        prov:
          format: int32
          type: integer_
"

**OAS with error**;
_    properties:
        id:
          allOf:
          - $ref: '#/components/schemas/UUID'
          - readOnly: true
        prov:
          format: int32
          type: integer_

**Portman version**;
```
npm list json-schema-merge-allof
<user>
└─┬ @apideck/portman@1.20.1
  └─┬ openapi-to-postmanv2@2.14.1
    └── json-schema-merge-allof@0.8.1
```
``` npm list openapi-to-postmanv2
<user>
└─┬ @apideck/portman@1.20.1
  └── openapi-to-postmanv2@2.14.1

```

[oas_subtest.yaml.txt](https://github.com/apideck-libraries/portman/files/10227359/oas_subtest.yaml.txt)
[oas_subtest_allof.yaml.txt](https://github.com/apideck-libraries/portman/files/10227360/oas_subtest_allof.yaml.txt)
[portman_error_msg.txt](https://github.com/apideck-libraries/portman/files/10227408/portman_error_msg.txt)

**Comment by thim81** [comment, marked as answer]
@pongohaja We have plans to upgrade to the latest openapi-to-postman, but since it also introduces support for OAS 3.0 & 3.1, we want to make sure our tests are all green and we can provide some basic support for OAS 3.1

**Reply by pongohaja** [reply]
@thim81 Thanks for you swift repy. Can you give a rough estimation of when it will be released? Next month or perhaps one year ahead?
